### PR TITLE
session,domain,owner: test the session package with tikv

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Make the session package testable with the real TiKV, so we can cover more transaction test cases.

```
make failpoint-enable
cd session
go test -with-tikv -pd-addrs=127.0.0.1:2379
```

### What is changed and how it works?

* add `withTiKV` flag and let the test suite check it
* call owner.Cancel() in domain so as to clean up the etcd registry
* reset `domap` and `storeBootstrapped` global variable in each SetupSuite()

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @coocood @jackysp

File: domain/domain.go [hunk 1]
```diff
@@ -509,6 +509,12 @@ func (do *Domain) loadSchemaInLoop(lease time.Duration) {
 		case <-do.exit:
 			return
 		}
+
+		select {
+		case <-do.exit:
```
Comment (coocood) [inline domain/domain.go]: We select `case <-do.exit` already in the above code.
Why do another select?

File: owner/manager.go [hunk 2]
```diff
@@ -235,6 +235,10 @@ func (m *ownerManager) campaignLoop(ctx context.Context, etcdSession *concurrenc
 		}
 
 		select {
+		case <-ctx.Done():
```
Comment (coocood) [inline owner/manager.go]: Why change the select order?

File: owner/manager.go [hunk 3]
```diff
@@ -279,6 +279,13 @@ func (m *ownerManager) campaignLoop(ctx context.Context, etcdSession *concurrenc
 
 		metrics.CampaignOwnerCounter.WithLabelValues(m.prompt, metrics.NoLongerOwner).Inc()
 		logutil.Logger(logCtx).Warn("is not the owner")
+
+		select {
+		case <-ctx.Done():
```
Comment (coocood) [inline owner/manager.go]: This seems doesn't make any difference.

File: session/pessimistic_test.go [hunk 4]
```diff
@@ -582,7 +562,11 @@ func (s *testPessimisticSuite) TestWaitLockKill(c *C) {
 	_, err := tk2.Exec("update test_kill set c = c + 1 where id = 1")
 	wg.Done()
 	c.Assert(err, NotNil)
-	c.Assert(terror.ErrorEqual(err, tikv.ErrQueryInterrupted), IsTrue)
+	if *withTiKV {
+		// Test not stable under TiKV
```
Comment (coocood) [inline session/pessimistic_test.go]: Do you know the reason?

Comment (tiancaiamao) [reply session/pessimistic_test.go]: Fixed.
The wait time is not long enough, so both `ErrQueryInterrupted` and `ErrLockWaitTimeout` are triggered. We check `ErrLockWaitTimeout` before `ErrQueryInterrupted`
https://github.com/pingcap/tidb/blob/master/store/tikv/2pc.go#L767

**Comment by tiancaiamao** [comment]
I reset the etcd status in SetupSuite.
PTAL @coocood

**Comment by coocood** [comment]
LGTM

File: store/tikv/2pc.go [hunk 5]
```diff
@@ -764,7 +764,7 @@ func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *
 			} else {
 				// the lockWaitTime is set, we should return wait timeout if we are still blocked by a lock
 				if time.Since(lockWaitStartTime).Milliseconds() >= action.LockWaitTime {
-					return ErrLockWaitTimeout
+					return errors.Trace(ErrLockWaitTimeout)
```
Comment (lonng) [inline store/tikv/2pc.go]: The `ErrLockWaitTimeout` contains the stack information already.

File: store/tikv/2pc.go [hunk 6]
```diff
@@ -777,7 +777,7 @@ func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *
 			// actionPessimisticLock runs on each region parallelly, we have to consider that
 			// the error may be dropped.
 			if atomic.LoadUint32(action.Killed) == 1 {
-				return ErrQueryInterrupted
+				return errors.Trace(ErrQueryInterrupted)
```
Comment (lonng) [inline store/tikv/2pc.go]: ditto

File: session/tidb.go [hunk 7]
```diff
@@ -98,6 +98,15 @@ func (dm *domainMap) Delete(store kv.Storage) {
 	dm.mu.Unlock()
 }
 
+// ResetForWithTiKVTest is only used in the test code.
+// TODO: Remove domap and storeBootstrapped. Use store.SetOption() to do it.
+func ResetForWithTiKVTest() {
```
Comment (lonng) [inline session/tidb.go]: It's better to place this function into `_test.go` and make it invisible for normal code.

File: store/tikv/2pc.go (hunk 5 above)
Comment (tiancaiamao) [reply store/tikv/2pc.go]: No, it doesn't.
Here it's not `ErrLockWaitTimeout.GenWithStack()`

File: session/tidb.go (hunk 7 above)
Comment (tiancaiamao) [reply session/tidb.go]: Done

**Pull request merged**